spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8SecretsPropertySource.java

```diff
@@ -40,40 +42,40 @@
 
 	public Fabric8SecretsPropertySource(KubernetesClient client, String name, String namespace,
 			Map<String, String> labels, boolean failFast) {
-		super(getSourceName(name, namespace), getSourceData(client, name, namespace, labels, failFast));
+		super(getSourceName(name, getApplicationNamespace(client, namespace, "Secret", null)),
+			getSourceData(client, name, getApplicationNamespace(client, namespace, "Secret", null), labels, failFast));
 	}
 
 	private static Map<String, Object> getSourceData(KubernetesClient client, String name, String namespace,
 			Map<String, String> labels, boolean failFast) {
 		Map<String, Object> result = new HashMap<>();
 
-		String namespaceToUse = StringUtils.hasLength(namespace) ? namespace : client.getNamespace();
```

Comment:
we do not need this line anymore, as it already has happened earlier. If you have reached this point a namespace is already known